lambdas/backend-client/src/client.ts

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios';
+import { ITemplateClient } from './types/template-client';
+import {
+  CreateTemplate,
+  Failure,
+  Success,
+  SuccessList,
+  UpdateTemplate,
+} from './types/generated';
+
+type ApiResponse = Failure | Success | SuccessList;
+
+const client = axios.create({
+  baseURL: process.env.TEMPLATE_API_URL,
+});
+
+const failure = (data: Failure) => ({
+  error: {
+    code: data.statusCode,
+    message: data.technicalMessage,
+    details: data.details,
+  },
+});
+
+export const templateClient: ITemplateClient = {
```

Comment:
this whole file is just an experiment/example (hence no unit tests). The intention behind `ITemplateClient` is both the backend-api and the web-ui both implement the interface so that if any types, functions or signatures change then it'll break both sides.